### DISCUSSION TITLE
RIS question

### DISCUSSION THREAD

**Discussion by tlstkddn999** [NVlabs/sionna#515] [Q&A] [answered]
Hello everyone

Here are my questions about handling RIS (Reconfigurable Intelligent Surfaces):

1. It seems that the path through the RIS only calculates the LOS (Line of Sight) path between TX - RIS and RIS - RX. Is this really how it is applied? Or, as with path calculation, are paths due to reflection or diffraction also included in the calculation?

<img width="409" alt="image" src="https://github.com/user-attachments/assets/f785cf05-b8b0-4da0-a318-38652500d9d7">

2. Similar to question 1, if multipath paths between RIS, TX, and RX can be calculated, is it possible to obtain values such as 'a', 'tau', ZOA, ZOD, etc., for those paths? It appears that the paths calculated through compute_path only include information about TX-RIS-RX, not TX-RIS or RX-RIS.

3. After obtaining the phase pattern for specific TX and RX positions through the focusing_lens function of RIS, if the RIS steers the beam to the RX position with that pattern, does it consider the situation where the rays coming out from RIS also undergo reflection or diffraction along different paths?
In other words, do beams coming out from RIS also experience reflection or diffraction like the rays from TX?

<img width="394" alt="image" src="https://github.com/user-attachments/assets/b09dfcf1-2763-44e6-bbd5-2983d774c665">

4. Lastly, when considering a multi-RIS scenario, even though higher gains can be confirmed on the coverage map, it seems that the corresponding paths are not calculated through compute_path. Is this part not supported yet?

<img width="658" alt="image" src="https://github.com/user-attachments/assets/c49495b8-ed09-46c2-bf83-3b3e88da3fc5">

Thank you :)

**Comment by jhoydis** [comment, marked as answer]
Hi @tlstkddn999,

First of all, you must understand that the method ``scene.compute_path()`` only considers paths of the type TX-RIS-RX.
The method ``scene.coverage_map()`` does not have this restriction. 

1) Yes. Only paths of the type TX-RIS-RX are considered for ``compute_paths``.

2) We typically do not return this information for all intersection points as it would quickly explode the memory. You could compute these values yourself if you want from the coordinates of the intersection points on the RIS.

3) No it does not. ``focusing_lens()` is a helper function that tries to create constructive interference from all unit cells at a particular point. As mention above, for the exact path computation only paths of the type TX-RIS-RX are considered.

4) Again, as mentioned above, paths of the type TX-RIS-RIS-RX or something similar are not supported in ``scene.compute_path()``.

**Reply by tlstkddn999** [reply]
@jhoydis 
thank you for the quick response, it was very helpful!
